server/service/orbit_client.go

```diff
@@ -338,3 +338,15 @@ func OrbitRetryInterval() time.Duration {
 	}
 	return constant.OrbitEnrollRetrySleep
 }
+
+// SetOrUpdateDiskEncryptionPayload sends a request to the server to set or update the disk
+// encryption keys and result of the encryption process
+func (oc *OrbitClient) SetOrUpdateDiskEncryptionPayload(diskEncryptionStatus fleet.OrbitHostDiskEncryptionKeyPayload) error {
+	verb, path := "POST", "/api/fleet/orbit/disk_encryption_key"
+
+	var resp setOrUpdateDiskEncryptionResponse
+	if err := oc.authenticatedRequest(verb, path, &diskEncryptionStatus, &resp); err != nil {
```

Comment:
To properly pass the right JSON payload, you should transfer from your `OrbitHostDiskEncryptionKeyPayload` struct to the request struct for this endpoint when integrating with my PR, it is here: https://github.com/fleetdm/fleet/pull/14004/files#diff-6ae374180fab3c63484cf9f776cbd09ab34b425eea5fcd79395a23446b6c1a16R459 (this will ensure the right field names, and the proper base64 encoding for the encryption key as it is a `[]byte` field).

Reply:
Done